packages/grid/src/vaadin-grid-drag-and-drop-mixin.js

```diff
@@ -257,6 +257,10 @@ export const DragAndDropMixin = (superClass) =>
           // The dragover didn't occur on a body row but the grid has items
           if (row) {
             // The dragover occurred over a header/footer row
+            if (this.__draggedItems && this.__draggedItems.length > 0) {
+              e.preventDefault();
+              e.dataTransfer.dropEffect = 'none';
+            }
```

Comment:
I don't see the reason for explicitly preventing dropping item data on header inputs as long as there's no exception.

https://github.com/user-attachments/assets/175a5118-ff3c-4a50-a53b-147c02a9e9fb

Reply:
That's a good question. I assumed that in the context of a grid, the header row of the grid would be considered wholly (including an input) as a target or not for a grid row.

Reply:
A grid row can basically be dropped anywhere as it contains (customizable) transfer data. We shouldn't interfere with that. Only when a row (or something else) is dropped over a grid's valid drop target (depending on the config), grid will handle the drop and dispatch the "grid-drop" event.